### PR TITLE
adding user_credentials grant type 

### PR DESCRIPTION
**Pull request by gambol99**
- adding the user_credentials grant type

### CODE REVIEW

**Comment by ericchiang** [comment]
As a note, grant_type=password is an OAuth2 feature not used by OpenID Connect. Are you using the oauth2 package on it's own, or are you attempting to interact with an OpenID Connect provider?

**Comment by gambol99** [comment]
Apologizes, I wasn't aware of that. Were using library to interact with [Keycloak](http://keycloak.jboss.org/) service.

File: oauth2/oauth2.go [hunk 1]
```diff
@@ -225,6 +226,30 @@ func (c *Client) ClientCredsToken(scope []string) (result TokenResponse, err err
 	return parseTokenResponse(resp)
 }
 
+// UserCredsToken posts the username, password, client_id and optional client_secret depending if client is public or not.
+// May not be supported by all OAuth2 servers.
+func (c *Client) UserCredsToken(username, password string, scope []string) (result TokenResponse, err error) {
```
Comment (ericchiang) [inline oauth2/oauth2.go]: Scopes are already on the Client struct, please use those.

**Comment by gambol99** [comment]
@ericchiang ... removed the param and using the c.scope

**Comment by ericchiang** [comment]
lgtm

**Pull request merged**